satellite/overlay/internaldbcache.go

```diff
@@ -0,0 +1,141 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package overlay
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/storj"
+)
+
+type addressInfo struct {
+	address   string
+	lastIP    string
+	transport pb.NodeTransport
+}
+
+type uptimeInfo struct {
+	isUp       bool
+	lastUptime time.Time
+	stats      *NodeStats
+}
+
+// InternalDBCache is a simple caching mechanism for overlaycache updates. It
+// provdes methods to help reduce calls to UpdateAddress and UpdateTime, but can
+// be extended for other calls in the future.
+type InternalDBCache struct {
```

Comment:
Having `Cache` and `InternalDBCache` will be confusing.

Reply:
I renamed it to `overlay.CombinedCache` I'm hoping this passes muster.